peachjam/admin.py

```diff
@@ -492,14 +499,25 @@ def reextract_content(self, request, queryset):
     reextract_content.short_description = "Re-extract content from DOCX files"
 
     def reindex_for_search(self, request, queryset):
-        """Setup a background task to re-index documents for search."""
+        """Set up a background task to re-index documents for search."""
         count = queryset.count()
         for doc in queryset:
             search_model_saved(doc._meta.label, doc.pk)
         self.message_user(request, f"Queued tasks to re-index for {count} documents.")
 
     reindex_for_search.short_description = "Re-index for search (background)"
 
+    def apply_labels(self, request, queryset):
+        """Set up a background task to apply labels to documents."""
+        count = queryset.count()
+        for doc in queryset:
+            apply_labels(doc.pk)
```

Comment:
i suggest just applying the labels directly -- it's not a risky or long-running call